airflow/utils/session.py

```diff
@@ -24,12 +24,17 @@
 from sqlalchemy.orm import Session as SASession
 
 from airflow import settings
+from airflow.api_internal.internal_api_call import InternalApiConfig
+from airflow.settings import TracebackSession
 from airflow.typing_compat import ParamSpec
 
 
 @contextlib.contextmanager
 def create_session() -> Generator[SASession, None, None]:
     """Contextmanager that will create and teardown a session."""
+    if InternalApiConfig.get_use_internal_api():
```

Comment:
should be:

```
if conf.getboolean("core", "database_access_isolation", fallback=False)
```

Reply:
Why?
I thought that `database_access_isolation` might be true globally for the deployment but e.g. scheduler and webserver would disregard that and still connect to DB?  I.e. with `force_database_direct_access()`?

If that's not true, perhaps instead of `force_database_direct_access` we should just throw an error if scheduler / webserver have `database_access_isolation` True?

Reply:
Ah yes. Now I re-read it, it's been a while.